x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/swimlane/steps/swimlane_connection.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiFieldText,
+  EuiFormRow,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import React, { useCallback, useMemo } from 'react';
+import { FormattedMessage } from 'react-intl';
+import * as i18n from '../translations';
+import { StepProps } from './';
+import { useKibana } from '../../../../../common/lib/kibana';
+import { getApplication } from '../api';
+
+export const SwimlaneConnection: React.FunctionComponent<StepProps> = ({
+  action,
+  editActionConfig,
+  editActionSecrets,
+  errors,
+  readOnly,
+  updateCurrentStep,
+  updateFields,
+}) => {
+  const { http } = useKibana().services;
+  const { apiUrl, appId } = action.config;
+  const { apiToken } = action.secrets;
+  const { docLinks } = useKibana().services;
+  const isValid = useMemo(() => apiUrl && apiToken && appId, [apiToken, apiUrl, appId]);
+  const connectSwimlane = useCallback(async () => {
+    // fetch swimlane application configuration
+    const application = await getApplication({ http, url: apiUrl, appId, apiToken });
```

Comment:
I am skeptical about doing the call on the front-end. The user should have had to allow the Kibana's instance url in the CORS policies which is maybe not be desired. Moving to the backend make more sense to me.

Reply:
im not against that

Reply:
Nevermind, we cannot move it to the backend as the connector has not been created yet and we cannot `_execute` 😄